docs/ci-cd-pipeline-assessment.md

```
@@ -0,0 +1,391 @@
+### 2.1 CI Steps
+
+| Language | CI Approach | Reusable Workflow |
+|----------|-----------|-------------------|
+| Go | Lint (golangci-lint), format check, tests with coverage, Trivy security scan | `go-ci.yml` |
+| Python (Docker repos) | Inline in `main.yml` — pip install + unittest | None (inline) |
+| Rust | None | None |
+| Kotlin | None | None |
+
+### 2.2 CD Steps — Go Lambda (S3 zip)
+
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)